src/com/rcv/ResultsWriter.java

```diff
@@ -0,0 +1,228 @@
+ * Currently we support an xlsx spreadsheet which can be visualized in a web browser
+ *
+ */
+public class ResultsWriter {
+
+  // takes tabulation round tallies and generates a spreadsheet from them
+  public boolean generateSummarySpreadsheet(
+      int finalRound,
+      Map<Integer, Map<String, Integer>> roundTallies,
+      Map<String, Integer> candidatesToRoundEliminated,
+      String outputFilePath
+  ) {
+    XSSFWorkbook workbook = new XSSFWorkbook();
+    XSSFSheet worksheet = workbook.createSheet("Election Results");
+
+    // TODO: write the election header info here (contest name, total votes cast, threhold, etc..)
+
+    /////////////////////////////////////
+    // Round-by-round elections results
+    /////////////////////////////////////
+
+    // Row 0: Round headers
+    int rowCounter = 0;
+    org.apache.poi.ss.usermodel.Row headerRow = worksheet.createRow(rowCounter++);
+    for(int round = 1; round <= finalRound; round++) {
+    // cellIndex is (round - 1) because rounds are 1-based while cell is 0-based
+    // cellIndex is multiplied by two because for each round we create two entries
+    // cellIndex is offset by 1 to account for row headers
+      int cellIndex = ((round-1)*2)+1;
+      Cell roundTotalCell = headerRow.createCell(cellIndex++);
+      String roundTotalText = String.format("Round %d total",round);
+      roundTotalCell.setCellValue(roundTotalText);
+      String roundDeltaText = String.format("Round %d delta",round);
+      Cell roundDeltaCell = headerRow.createCell(cellIndex);
+      roundDeltaCell.setCellValue(roundDeltaText);
+    }
+
+    // Row 1: Defeated candidate lists
+
+    // invert candidatesToRoundEliminated so we can lookup who got eliminated for each round
+    Map<Integer, List<String>> roundToCandidatesEliminated = new HashMap<Integer, List<String>>();
+    for (String candidate : candidatesToRoundEliminated.keySet()) {
+      int round = candidatesToRoundEliminated.get(candidate);
+      if (roundToCandidatesEliminated.get(round) == null) {
+        roundToCandidatesEliminated.put(round, new LinkedList<String>());
+      }
+      roundToCandidatesEliminated.get(round).add(candidate);
+    }
+
+    // show eliminations for each round
+    StringBuilder sb = new StringBuilder("Eliminations: ");
+    org.apache.poi.ss.usermodel.Row eliminationsRow = worksheet.createRow(rowCounter++);
+    Cell eliminationsRowHeader = eliminationsRow.createCell(0);
+    eliminationsRowHeader.setCellValue("eliminations");
+    for (int round = 1; round <= finalRound; round++) {
+      sb.append(round).append(": ");
+      List<String> eliminated = roundToCandidatesEliminated.get(round);
+      if (eliminated != null) {
+        String cellText = String.join(", ", eliminated);
+        int cellIndex = ((round-1)*2)+1;
+        Cell cell = eliminationsRow.createCell(cellIndex);
+        cell.setCellValue(cellText);
+        sb.append(cellText);
+      } else {
+        sb.append("none");
```

Comment:
Should this be possible? Seems like an error condition.

Reply:
if we stop at the penultimate round then this would not happen.  if we have a column for the final round (as I would prefer, just for simplicity) then that round will have no eliminations

Reply:
So in the penultimate round, the runner-up is eliminated? What happens in the final round? (How is it different from the penultimate round?)

Reply:
no, I got that wrong.  there should still be an elimination(s) in each round.  there's a bug where the final elimination isn't recorded by the tabulator (the main loop just exits) and so the final elimination isn't written to the spreadsheet

Reply:
Ah. Are you going to send out a PR for the bug fix? Or I can take a look later.

Reply:
Just pushed a fix to this same branch.

Reply:
Thanks. Seems legit.